### PR TITLE
provider/ignition: migration from resources to data resources

### PR DESCRIPTION
**Pull request by mcuadros**
Fixes: https://github.com/hashicorp/terraform/issues/11518

### CODE REVIEW

**Comment by stack72** [comment]
Hi @mcuadros 

Thanks for the work here - I understand why we need to do this. Right now, this is backwards incompatible - which means we would need to duplicate the resources and deprecate the old ones etc

Luckily, the awesome @apparentlymart, wrote a shim function that he has used to map resources to data_sources

You can see an example of it [here](https://github.com/hashicorp/terraform/blob/master/builtin/providers/terraform/provider.go#L9)
and [here](https://github.com/hashicorp/terraform/blob/master/builtin/providers/archive/provider.go#L9)

This would then allow us to keep backwards compatibility with older versions :)

Thoughts?

Paul

**Comment by mcuadros** [comment]
@stack72 done

**Comment by mcuadros** [comment]
@stack72 ping

File: website/source/docs/providers/ignition/index.html.markdown [hunk 1]
```diff
@@ -19,6 +19,19 @@ Use the navigation to the left to read about the available resources.
 This config will write a single service unit (shown below) with the contents of an example service. This unit will be enabled as a dependency of multi-user.target and therefore start on boot
 
 ```
+<<<<<<< HEAD
```
Comment (mildred) [inline website/source/docs/providers/ignition/index.html.markdown]: Is this a merge that failed?

File: website/source/docs/providers/ignition/index.html.markdown [hunk 2]
```diff
@@ -19,6 +19,19 @@ Use the navigation to the left to read about the available resources.
...
+	name = "example.service"
+	content = "[Service]\nType=oneshot\nExecStart=/usr/bin/echo Hello World\n\n[Install]\nWantedBy=multi-user.target"
+}
+
+# Ingnition config include the previous defined systemd unit data resource
+data "ignition_config" "example" {
+	systemd = [
+		"${data.ignition_systemd_unit.example.id}",
+	]
+=======
```
Comment (mildred) [inline website/source/docs/providers/ignition/index.html.markdown]: merge conflict?

**Comment by mcuadros** [comment]
@mildred nice catch! solved

**Comment by pgray** [comment]
Pulled down this branch and started working with it.
Great work!

**Comment by mildred** [comment]
By the way, we are using this in production and it works well.

**Comment by alexsomesan** [comment]
👍 I would also like to see this merged. It's a biting us a little bit, using the latest stable release.

**Comment by stack72** [comment]
LGTM! Thanks for the work here - sorry it took so long 

```
% make testacc TEST=./builtin/providers/ignition                                                                                            ✭
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/03/06 14:21:46 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/ignition -v  -timeout 120m
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestValidateUnit
--- PASS: TestValidateUnit (0.00s)
=== RUN   TestIngnitionFileReplace
--- PASS: TestIngnitionFileReplace (0.05s)
=== RUN   TestIngnitionFileAppend
--- PASS: TestIngnitionFileAppend (0.04s)
=== RUN   TestIngnitionDisk
--- PASS: TestIngnitionDisk (0.07s)
=== RUN   TestIngnitionDiskResource
--- PASS: TestIngnitionDiskResource (0.30s)
=== RUN   TestIngnitionFile
--- PASS: TestIngnitionFile (0.06s)
=== RUN   TestIngnitionFilesystem
--- PASS: TestIngnitionFilesystem (0.07s)
=== RUN   TestIngnitionGroup
--- PASS: TestIngnitionGroup (0.04s)
=== RUN   TestIngnitionNetworkdUnit
--- PASS: TestIngnitionNetworkdUnit (0.03s)
=== RUN   TestIngnitionRaid
--- PASS: TestIngnitionRaid (0.04s)
=== RUN   TestIngnitionSystemdUnit
--- PASS: TestIngnitionSystemdUnit (0.03s)
=== RUN   TestIngnitionSystemdUnitEmptyContentWithDropIn
--- PASS: TestIngnitionSystemdUnitEmptyContentWithDropIn (0.03s)
=== RUN   TestIgnitionSystemdUnit_emptyContent
--- PASS: TestIgnitionSystemdUnit_emptyContent (0.03s)
=== RUN   TestIngnitionUser
--- PASS: TestIngnitionUser (0.03s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/ignition	0.851s
```

**Pull request merged**